### DISCUSSION TITLE
GitHub repository setup

### DISCUSSION THREAD

**Discussion by gucio321** [AllenDang/cimgui-go#50] [Ideas]
Hi @AllenDang 
thank you for inviting me to collaborating on this repository!

I've a proposal about setting up a `branch-protection-rules`. 
In my opinion the protection rule should now enforce at least one PR review. This way, one person would give approval, and another one will merge. This will reduce a possibility of merging bugged code to master :smile:

we should also add some auto-tests. e.g. a simple workflow file that will run `go test ./...` to ensure the code isn't buildable

**Comment by gucio321** [comment]
I'd blame #18

**Comment by AllenDang** [comment]
Totally agree

**Comment by gucio321** [comment]
> I'd blame https://github.com/AllenDang/cimgui-go/pull/18

ops, sorry, wrong issue :smile:

> Totally agree

great!, Sadly, I can't do that, since I don't have access to settings, but the setup should look like that:
![image](https://user-images.githubusercontent.com/73652197/205452033-d5673fe9-f8c6-4097-bcdd-b0a02b540685.png)

**Reply by gucio321** [reply]
very well, so since #45 is already merged, @AllenDang you need to enable rule for this check
and could you also check this "Require branches to be up-to-data..." - it makes "Update branch" button appear and it's easier to update branch fro github's UI :smile:
![image](https://user-images.githubusercontent.com/73652197/207096651-36a63c62-b82d-47e1-9064-0e5498d5c00c.png)

**Comment by eliasdaler** [comment]
Another proposal: let's enable "Discussions" in this repo and move topics like these there instead of making many "issues" as they're usually for bugs/feature requests, etc.

**Comment by gucio321** [comment]
> Another proposal: let's enable "Discussions" in this repo and move topics like these there instead of making many "issues" as they're usually for bugs/feature requests, etc.

it is done now. Converting this to discussion

**Comment by AllenDang** [comment]
@gucio321 I just created the branch protection rules

**Reply by gucio321** [reply]
asesome!

**Comment by gucio321** [comment]
Hi!
according to #105 we need to extend our test workflow to windows, since (like we saw) the code may compile on ubuntu, but may crash on windows
